pgscatalog.calc/src/pgscatalog/calc/cli/aggregate_cli.py

```diff
@@ -21,15 +21,41 @@ def run_aggregate():
 
     if args.verbose:
         logger.setLevel(logging.INFO)
+        logging.getLogger("pgscatalog.core").setLevel(logging.INFO)
+        logging.getLogger("pgscatalog.calc").setLevel(logging.INFO)
 
     if not (outdir := pathlib.Path(args.outdir)).exists():
         raise FileNotFoundError(f"--outdir {outdir.name} doesn't exist")
 
-    score_paths = [pathlib.Path(x) for x in args.scores]
-    pgs = [PolygenicScore(path=x) for x in score_paths]
-    # call __add__ a lot
-    aggregated = functools.reduce(operator.add, pgs)
+    score_paths = sorted([pathlib.Path(x) for x in args.scores], key=chrom_keyfunc())
+    # dfs are only read into memory after accessing them explicitly e.g. pgs[0].df
+    pgs = deque(PolygenicScore(path=x) for x in score_paths)
+
+    observed_columns = set()
+    aggregated = None
+
+    while pgs:
+        x = pgs.popleft()  # to remove dfs from memory during iteration
+        if aggregated is None:
+            logger.info(f"Initialising aggregation with {x}")
+            aggregated = x
+        else:
+            logger.info(f"Adding {x}")
+            aggregated += x
```

Comment:
Didn't we use df.add() before in pandas? Is this the same? Also, don't we need to recalculate the AVG from the SUM and DENOM?

Reply:
(we chatted on slack but for context re: `df.add`)

https://github.com/PGScatalog/pygscatalog/blob/8e389f6a5ee892573e6a6d80d5ecaebbb127246a/pgscatalog.calc/src/pgscatalog/calc/lib/polygenicscore.py#L339-L345

the latest commit hopefully improves the logic for averaging, melting, and writing out, which is too abstract